src/wazuh_db/wdb_syscollector.c

```diff
@@ -234,12 +245,17 @@ int wdb_netaddr_save(wdb_t * wdb, const char * scan_id, const char * iface, int
 
 // Insert IPv4/IPv6 address info tuple. Return 0 on success or -1 on error.
 int wdb_netaddr_insert(wdb_t * wdb, const char * scan_id, const char * iface, int proto, const char * address, const char * netmask, const char * broadcast, const char * checksum, const char * item_id, const bool replace) {
-
     sqlite3_stmt *stmt = NULL;
 
+    if (NULL == iface || NULL == address) {
+        if(checksum && 0 != strcmp(SYSCOLLECTOR_LEGACY_CHECKSUM_VALUE, checksum)) {
+            wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETADDRESS, item_id);
+        }
```

Comment:
```suggestion
        wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETADDRESS, item_id);
```
the item_id is a sync field, no need to re-identify it, check the call tree. (For legacy code, item_id is null)

Reply:
ditto